include/openenclave/internal/plugin.h

```diff
@@ -0,0 +1,465 @@
+     * @param[in] config_data An optional pointer to the configuration data.
+     * @param[in] config_data_size The size in bytes of config_data.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*on_register)(
+        oe_attestation_role_t* context,
+        const void* config_data,
+        size_t config_data_size);
+
+    /**
+     * The function that gets executed when the attestation role is
+     * unregistered.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attestation role struct.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*on_unregister)(oe_attestation_role_t* context);
+};
+
+#ifdef OE_BUILD_ENCLAVE
+
+/**
+ * The attester attestion role. The attester is reponsible for generating the
+ * attestation evidence and must implement the functions below.
+ */
+typedef struct _oe_attester oe_attester_t;
+struct _oe_attester
+{
+    /**
+     * The base attestation role containing the common functions for each role.
+     */
+    oe_attestation_role_t base;
+
+    /**
+     * Generates the attestation evidence, which is defined as the data
+     * produced by the enclave. The caller may pass in custom claims, which
+     * must be attached to the evidence and then cryptographically signed.
+     *
+     * Note that many callers of `get_evidence` will send the results over
+     * the network, so the output must be in a serialized form.
+     *
+     * @experimental
+     *
+     * @param[in] context A pointer to the attester plugin struct.
+     * @param[in] custom_claims The optional custom claims list.
+     * @param[in] custom_claims_length The number of custom claims.
+     * @param[in] opt_params The optional plugin-specific input parameters.
+     * @param[in] opt_params_size The size of opt_params in bytes.
+     * @param[out] evidence_buffer An output pointer that will be assigned the
+     * address of the evidence buffer.
+     * @param[out] evidence_buffer_size A pointer that points to the size of the
+     * evidence buffer in bytes.
+     * @param[out] endorsements_buffer An output pointer that will be assigned
+     * the address of the endorsements buffer.
+     * @param[out] endorsements_buffer_size A pointer that points to the size of
+     * the endorsements buffer in bytes.
+     * @retval OE_OK on success.
+     * @retval An appropriate error code on failure.
+     */
+    oe_result_t (*get_evidence)(
+        oe_attester_t* context,
+        const oe_claim_t* custom_claims,
+        size_t custom_claims_length,
+        const void* opt_params,
+        size_t opt_params_size,
+        uint8_t** evidence_buffer,
+        size_t* evidence_buffer_size,
+        uint8_t** endorsements_buffer,
+        size_t* endorsements_buffer_size);
+
+    /**
+     * Creates a legacy report to be used in local or remote attestation.
```

Comment:
```suggestion
     * Creates a legacy SGX report to be used in local or remote attestation.
```

Reply:
The comments for the legacy API `oe_get_report_v2()` in header files `include/openenclave/host.h` and `include/openenclave/enclave.h` do not explicitly state the output report to be SGX specific. I tried to be consistent here, though my understanding is that the legacy API functions `oe_get_report()` and `oe_verify_report()` only support SGX reports.

Reply:
right they do only support SGX, and the term "report" is an SGX-specific term in this context.